### PR TITLE
Clients

### PR DESCRIPTION
**Pull request by heron2014**
## Ready!

Related to this [326](https://github.com/FAC-GM/app/issues/326)

At the moment we have 2 endpoints for create and update client. Do you think we should handle these two endpoints as one only?

@nelsonic please can you review this PR? thank you.

### CODE REVIEW

File: lib/handlers/clients/post_delete.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+const clientES = require('../../es.js');
+
+module.exports = function (request, reply) {
+
+  if (!request.auth.isAuthenticated) {
+    return reply.redirect('/login');
+  } else {
+
+    clientES.delete({
```
Comment (nelsonic) [inline lib/handlers/clients/post_delete.js]: deleting data directly is a bad idea in a CRM. 
we should have a "soft delete" method which puts the client record in a recycling bin...
@Adam-JF unless we are *restricting* delete operations to only a few people, 
its quite probable that something will be *accidentally* deleted without any possibility of recovery...

**Comment by iteles** [comment]
Our conversations this afternoon reflect this concern, I think we agreed to replace this with an 'archive' functionality, but quite late on in the day already!

File: lib/handlers/clients/post_create.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  if (!request.auth.isAuthenticated) {
+    return reply.redirect('/login');
+  } else {
+
+    var schema = {
+      companyName: Joi.string().alphanum().min(2).required()
+    };
+
+    var validationObject = Joi.validate(request.payload, schema, { abortEarly: false, allowUnknown: true });
```
Comment (heron2014) [inline lib/handlers/clients/post_create.js]: thanks @nelsonic I will look into that!

**Comment by heron2014** [comment]
thanks @nelsonic & @iteles, I will amend the code to take off delete option unless @Adam-JF decides to keep it.

**Comment by nelsonic** [comment]
@heron2014 the *simple* way of achieving the "soft delete" effect is:
+ instead of a `DELETE` query, have an update query where you set  `deleted:true`
+ in your SELECT (GET/READ) query, add the filter `deleted:false`

This way the (*soft*) "Deleted" items will not show up in the UI anymore but can still be retrieved if an "Admin" wants to recover them.

**Comment by heron2014** [comment]
thanks @nelsonic, we are working on this now, we will have properties : active: true, active:false link to the client.

**Comment by nelsonic** [comment]
Let me know when ready to review. :+1:

**Comment by SimonLab** [comment]
@nelsonic This PR is ready to ve reviewed

**Comment by nelsonic** [comment]
Looks good. We should look at getting the last 2% of coverage back when possible. :+1:

**Pull request merged**